src/udev/udev-builtin-net_driver.c

```diff
@@ -0,0 +1,39 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "alloc-util.h"
+#include "device-util.h"
+#include "errno-util.h"
+#include "ethtool-util.h"
+#include "fd-util.h"
+#include "log.h"
+#include "string-util.h"
+#include "udev-builtin.h"
+
+static int builtin_net_driver_set_driver(UdevEvent *event, int argc, char **argv, bool test) {
+        sd_device *dev = ASSERT_PTR(ASSERT_PTR(event)->dev);
+        _cleanup_close_ int ethtool_fd = -EBADF;
+        _cleanup_free_ char *driver = NULL;
+        const char *sysname;
+        int r;
+
+        r = sd_device_get_sysname(dev, &sysname);
+        if (r < 0)
+                return log_device_warning_errno(dev, r, "Failed to get sysname: %m");
+
+        r = ethtool_get_driver(&ethtool_fd, sysname, &driver);
+        if (IN_SET(r, -ENODEV, -EOPNOTSUPP)) {
+                log_device_debug_errno(dev, r, "Failed to get driver for '%s': %m", sysname);
```

Comment:
Hmm, if you check for these might be nice to output something more useful than the same as the generic message below...

maybe:

```c
if (ERRNO_IS_NOT_SUPPORTED(r)) {
        log_debug_debug_errno(dev, r, "Querying driver name via ethttool API is not supported by device '%s': %m", sysname);
        return 0;
} 
if (r == -ENODEV) {
        log_device_debug_errno(dev, r, "Device already vanished, ignoring.");
        return 0;
}
if (r < 0) 
         …